src/tests/token/erc20/common.cairo

```diff
@@ -2,48 +2,49 @@ use openzeppelin::tests::utils;
 use openzeppelin::token::erc20::ERC20Component::{Approval, Transfer};
 use openzeppelin::token::erc20::ERC20Component;
 use openzeppelin::utils::serde::SerializedAppend;
+use snforge_std::{SpyOn, EventSpy, EventAssertions};
 use starknet::ContractAddress;
 
 pub(crate) fn assert_event_approval(
-    contract: ContractAddress, owner: ContractAddress, spender: ContractAddress, value: u256
+    ref spy: EventSpy,
+    contract: ContractAddress,
+    owner: ContractAddress,
+    spender: ContractAddress,
+    value: u256
 ) {
-    let event = utils::pop_log::<ERC20Component::Event>(contract).unwrap();
     let expected = ERC20Component::Event::Approval(Approval { owner, spender, value });
-    assert!(event == expected);
-
-    // Check indexed keys
-    let mut indexed_keys = array![];
-    indexed_keys.append_serde(selector!("Approval"));
-    indexed_keys.append_serde(owner);
-    indexed_keys.append_serde(spender);
-    utils::assert_indexed_keys(event, indexed_keys.span())
+    spy.assert_emitted(@array![(contract, expected)]);
```

Comment:
Actually, I should have left a TODO, but this is not testing indexed keys as we did before, since with this approach removing a key attribute in an event won't make the test fail. I'm waiting for the next foundry version to add those, since I saw there's already some features in main that would make this easier.

Reply:
> this is not testing indexed keys as we did before, since with this approach removing a key attribute in an event won't make the test fail

This feels a little redundant tbh since `EventAssertion` checks keys and data, but I do see the benefit (especially in development) with manually checking keys